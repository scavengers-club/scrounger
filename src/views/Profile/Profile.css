.profile-header {
  font-family: 'Reenie Beanie', cursive;
  font-size: 200%;
  text-transform: uppercase;
  text-align: center;
}

.profile {
  display: flex;
  flex-direction: column;
  align-items: center;
}
