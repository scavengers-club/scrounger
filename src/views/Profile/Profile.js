import { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { getUserAdventures } from '../../services/adventures';
import AdventureList from '../../components/AdventureList/AdventureList';

export default function Profile({ currentUser }) {
  const [adventures, setAdventures] = useState([]);
  const [loading, setLoading] = useState(true);
  const userId = currentUser.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await getUserAdventures(userId);
      setAdventures(data);
      setLoading(false);
    };
    fetchData();
  }, [userId]);

  if (loading) return <h1>loading</h1>;

  return (
    <div>
      <h1>Hi I am Profile</h1>
      <AdventureList {...{ adventures }} />
      <Link exact to="/adventure/create">
        create new adventure
      </Link>
    </div>
  );
}
