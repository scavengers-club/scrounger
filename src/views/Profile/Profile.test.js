import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Profile from './Profile';

test('<Profile/> should render the users profile', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/profile']}>
      <Route path="/profile">
        <Profile currentUser={{ id: '575e91dd-2c9c-4297-aa60-6504a62236dc', username: 'violet' }} />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('Here are your adventures, violet');
  expect(container).toMatchSnapshot();
});
