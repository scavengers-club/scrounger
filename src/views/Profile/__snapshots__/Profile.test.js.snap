// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> should render the users profile 1`] = `
<div>
  <div
    class="profile"
  >
    <h1
      class="profile-header"
    >
      Here are your adventures, 
      violet
    </h1>
    <a
      href="/adventure/create"
    >
      create new adventure
    </a>
    <p>
      You have no adventures yet...
    </p>
    <div
      class="adventure-list"
    />
  </div>
</div>
`;
